Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -1598,15 +1601,7 @@ private bool IsMultiSelectAllIndeterminate
         /// Gets or sets the datagrid data-source.
         /// </summary>
         [Parameter]
-        public IEnumerable<TItem> Data
-        {
-            get { return data; }
-            set
-            {
-                SetDirty();
-                data = value;
-            }
-        }
+        public IEnumerable<TItem> Data { get; set; }

```

Comment:
I like this change.

Reply:
Right I think we should disregard doing logic in the parameter setters and move it to the SetParameters, where we even have async support.
I think there's even an analyser that throws a warning to not do stuff on blazor parameters.

Reply:
At least the logic that will trigger rerender. Other props that set `DirtyClasses()` are OK, I think.

Reply:
Actually I'd argue that those should also go there and it would even be more clear & concise what triggers DirtyClasses, since those would all be in the same place.

Reply:
Well, tbf, that was the idea for the new ComponentParameterInfo that is still in prototype limbo. I agree, we should refactor it slowly by moving to v2.